src/main/java/org/inferred/freebuilder/processor/DefaultPropertyFactory.java

```diff
@@ -169,18 +169,56 @@ public void addFinalFieldAssignment(SourceBuilder code, String finalField, Strin
 
     @Override
     public void addMergeFromValue(Block code, String value) {
-      code.addLine("%s(%s.%s());", setter(property), value, property.getGetterName());
+      Excerpt defaults = Declarations.freshBuilder(code, metadata).orNull();
```

Comment:
I don't know if this is a standard across the codebase, but I generally prefer `isPresent()` to check an optional value rather than `orNull` and then a null check.

You'd have to call `defaults.get()` all the time (the jvm should optimise this anyway), or store it in a variable, but I think it's still cleaner than having nullable variables. (Java 8's Optional `ifPresent(defaults -> )` would be convenient :) )